hnn_core/gui/gui.py

```diff
@@ -300,32 +251,24 @@ def _initialize_viz_window(simulation_data, analysis_config, init=False):
         'network',
     ]
     viz_window.clear_output()
-    previous_plot_outputs_values = []
-    while len(plot_outputs) > 0:
-        plot_outputs.pop()
-        previous_plot_outputs_values.append(plot_dropdowns.pop().value)
-    # keep the same order of previous plots.
-    previous_plot_outputs_values.reverse()
-
-    with viz_window:
-        # Left-Rright configuration
-        if layout_option == "L-R":
-            grid = _init_left_right_viz_layout(plot_outputs, plot_dropdowns,
-                                               simulation_data, plot_options,
-                                               previous_plot_outputs_values,
-                                               analysis_config, layout,
-                                               init=init)
+    previous_plot_types = {}
 
-        # Upper-Down configuration
-        elif layout_option == "U-D":
-            grid = _init_upper_down_viz_layout(plot_outputs, plot_dropdowns,
-                                               simulation_data, plot_options,
-                                               previous_plot_outputs_values,
-                                               analysis_config, layout,
-                                               init=init)
+    plot_keys = copy.deepcopy(list(plot_outputs.keys()))
+    for plot_key in plot_keys:
+        previous_plot_types[plot_key] = plot_dropdowns[plot_key].value
+        del plot_outputs[plot_key]
+        del plot_dropdowns[plot_key]
 
-        viz_grid[layout_option] = grid
-        display(grid)
+    with viz_window:
+        viz_grid[layout_option] = _init_viz_layout(
+            simulation_data,
+            plot_options,
+            previous_plot_types,
+            analysis_config,
+            layout,
+            layout_option,
```

Comment:
use keyword arguments to be clearer when there are many parameters to the function ... is it `layout` or `style`? Seems like you call it `style` inside the function

Reply:
Yeah there's some inconsistency... will fix later. Now I need to fix the GUI test first 😂

Reply:
Fixed in commit 0eb29c7a3118f0b2302abed83c3a2e567c4f15af